lib/KettleRouter.js

```diff
@@ -0,0 +1,108 @@
+/*
+Routing Primitives for Kettle Servers
+
+Copyright 2015 Raising the Floor (International)
+
+Licensed under the New BSD license. You may not use this file except in
+compliance with this License.
+
+You may obtain a copy of the License at
+https://github.com/fluid-project/kettle/blob/master/LICENSE.txt
+*/
+
+/* Contains code adapted from express 4.x "layer.js":
+ * Copyright(c) 2009-2013 TJ Holowaychuk
+ * Copyright(c) 2013 Roman Shtylman
+ * Copyright(c) 2014-2015 Douglas Christopher Wilson
+ * MIT Licensed
+ */
+
+"use strict";
+
+var fluid = require("infusion"),
+    urlModule = require("url"),
+    kettle = fluid.registerNamespace("kettle");
+
+// Upstream dependency stolen from express 4.x    
+kettle.pathToRegexp = require("path-to-regexp");
+    
+fluid.defaults("kettle.router.http", {
```

Comment:
Historically WebSockets requests were routed by a different variety of router... in the end it seemed much more logical to route them using the existing primitives and simply omit their "method". However, room for expansion is left should different kinds of routers seem desirable (that is, we find ourselves handling other kinds of requests) - https://github.com/amb26/kettle/blob/KETTLE-32/lib/KettleServer.js#L173